modules/core_api/fsw/inc/cfe_es_api_typedefs.h

```diff
@@ -0,0 +1,193 @@
+/*
+**  GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**  Copyright (c) 2006-2019 United States Government as represented by
+**  the Administrator of the National Aeronautics and Space Administration.
+**  All Rights Reserved.
+**
+**  Licensed under the Apache License, Version 2.0 (the "License");
+**  you may not use this file except in compliance with the License.
+**  You may obtain a copy of the License at
+**
+**    http://www.apache.org/licenses/LICENSE-2.0
+**
+**  Unless required by applicable law or agreed to in writing, software
+**  distributed under the License is distributed on an "AS IS" BASIS,
+**  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**  See the License for the specific language governing permissions and
+**  limitations under the License.
+*/
+
+/**
+ * @file
+ *
+ *  Purpose:
+ *	Unit specification for Executive Services library functions and macros.
+ *
+ *  References:
+ *     Flight Software Branch C Coding Standard Version 1.0a
+ *     cFE Flight Software Application Developers Guide
+ *
+ *	Notes:
+ *
+ */
+
+#ifndef CFE_ES_API_TYPEDEFS_H
+#define CFE_ES_API_TYPEDEFS_H
+
+/*
+** Includes
+*/
+#include "common_types.h"
+#include "cfe_es_extern_typedefs.h"
+
+/*
+** Note about reset type and subtypes:
+**
+** These values come from the PSP so the actual definition of these enumerations
+** was moved to the PSP header file <cfe_psp.h>.
+**
+** In the future the Electronic Data sheets (EDS) for PSP/ES
+** will define the exact values to use in telemetry messages.
+*/
+
+/*
+**  Reset types
+*/
+/** \name Reset Type extensions */
+/** \{ */
+#define CFE_ES_APP_RESTART CFE_PSP_RST_TYPE_MAX /**< Application only was reset (extend the PSP enumeration here) */
+/** \} */
+
+/*****************************************************************************/
+/*
+** Type Definitions
+*/
+
+/*
+** Entry Function Prototypes
+*/
+typedef void (*CFE_ES_TaskEntryFuncPtr_t)(void); /**< \brief Required Prototype of Task Main Functions */
+typedef int32 (*CFE_ES_LibraryEntryFuncPtr_t)(
+    CFE_ES_LibId_t LibId); /**< \brief Required Prototype of Library Initialization Functions */
+
+/**
+ * \brief Compatible typedef for ES child task entry point.
+ *
+ * All ES task functions (main + child) use the same entry point type.
+ */
+typedef CFE_ES_TaskEntryFuncPtr_t CFE_ES_ChildTaskMainFuncPtr_t;
```

Comment:
If all ES task functions (main + child according to the comments) use the same entry point type, couldn't we just define CFE_ES_TaskEntryFuncPtr_t here as "void (*CFE_ES_TaskEntryFuncPtr_t)(void)" and just replace instances of CFE_ES_ChildTaskMainFuncPtr_t with this type?

Reply:
Yes, but this was an existing type in the CFE API so it was kept the same name for backward compatibility.

Reply:
See https://github.com/nasa/cFE/issues/1304